src/main/java/io/pravega/connectors/flink/FlinkPravegaWriter.java

```diff
@@ -531,33 +539,103 @@ public void restoreState(List<UUID> list) throws Exception {
             // we can have more than one transaction to check in case of a scale-in event, or
             // for the reasons discussed in the 'notifyCheckpointComplete()' method.
 
-            if (list != null && list.size() > 0) {
-                for (UUID txnId : list) {
-                    if (txnId != null) {
-                        final Transaction<?> txn = pravegaWriter.getTxn(txnId);
-                        final Transaction.Status status = txn.checkStatus();
+            // we will create a writer per scope/stream from the pending transaction list and
+            // commit the transaction if it is still open
+
+            if (pendingTransactionList == null || pendingTransactionList.size() == 0) {
+                return;
+            }
+
+            Exception exception = null;
+
+            EventStreamWriter<T> restorePravegaWriter = null;
+
+            ClientFactory restoreClientFactory = null;
```

Comment:
I bet you can eliminate these local variable declarations using try-with-resources.
```
try(
  ClientFactory restoreClientFactory = ..., 
  EventStreamWriter<T> restorePravegaWriter = ...
) {
 ...
}
```